BambooHR/Injector/BambooHTTPRequestInjector.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * BambooHTTPRequest.php
+ * @author    Daniel Mason <danielm@moo.com>
+ * @copyright 2016 MOO
+ * @license   proprietary
+ * @see       https://www.moo.com
+ */
+
+namespace BambooHR\API\Injector;
+use BambooHR\API\BambooHTTPRequest;
+
+
+/**
+ * BambooHTTPRequest
+ * Allows injection of BambooHttpRequest with a default to fall back on
+ * @package BambooHR
+ */
+trait BambooHTTPRequestInjector
```

Comment:
what advantage do you get from making this a trait vs. just adding these methods/member variables to the class?